<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1268</width>
    <height>459</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QTextEdit" name="textEdit">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>50</y>
      <width>401</width>
      <height>351</height>
     </rect>
    </property>
    <property name="readOnly">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>410</x>
      <y>10</y>
      <width>281</width>
      <height>281</height>
     </rect>
    </property>
    <property name="title">
     <string>Работа с парами ключ-значение</string>
    </property>
    <widget class="QWidget" name="verticalLayoutWidget">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>20</y>
       <width>281</width>
       <height>261</height>
      </rect>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QLabel" name="label">
        <property name="text">
         <string>Узнать значения ключей key1, key2, ...</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="get_pairs"/>
      </item>
      <item>
       <widget class="QPushButton" name="get_pairs_btn">
        <property name="text">
         <string>Выполнить</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="verticalSpacer_3">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QLabel" name="label_3">
        <property name="text">
         <string>Задать значения ключей key1=val1, key2=val2, ...</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="set_pairs"/>
      </item>
      <item>
       <widget class="QPushButton" name="set_pairs_btn">
        <property name="text">
         <string>Выполнить</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_2">
    <property name="geometry">
     <rect>
      <x>720</x>
      <y>10</y>
      <width>231</width>
      <height>281</height>
     </rect>
    </property>
    <property name="title">
     <string>Работа с хэшами</string>
    </property>
    <widget class="QWidget" name="verticalLayoutWidget_2">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>20</y>
       <width>231</width>
       <height>261</height>
      </rect>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>Ключ key</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="hash_key"/>
      </item>
      <item>
       <widget class="QPushButton" name="get_hash_btn">
        <property name="text">
         <string>Узнать значение хэша по ключу key</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="verticalSpacer">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QLabel" name="label_6">
        <property name="text">
         <string>Значения key1=val1, key2=val2, key3=val3</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="set_hash"/>
      </item>
      <item>
       <widget class="QPushButton" name="set_hash_btn">
        <property name="text">
         <string>Задать значения хэша по ключу key</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_3">
    <property name="geometry">
     <rect>
      <x>970</x>
      <y>10</y>
      <width>281</width>
      <height>401</height>
     </rect>
    </property>
    <property name="title">
     <string>Работа со списками</string>
    </property>
    <widget class="QWidget" name="verticalLayoutWidget_3">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>20</y>
       <width>281</width>
       <height>381</height>
      </rect>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_3">
      <item>
       <widget class="QLabel" name="label_7">
        <property name="text">
         <string>Ключ key</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="list_key"/>
      </item>
      <item>
       <widget class="QPushButton" name="get_list">
        <property name="text">
         <string>Просмотреть список key</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="verticalSpacer_4">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QLabel" name="label_10">
        <property name="text">
         <string>Значение</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="list_val"/>
      </item>
      <item>
       <widget class="QPushButton" name="list_addl">
        <property name="text">
         <string>Добавить значение с начала списка</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="list_addr">
        <property name="text">
         <string>Добавить значение с конца списка</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="verticalSpacer_6">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeType">
         <enum>QSizePolicy::Fixed</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>24</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QPushButton" name="list_delr">
        <property name="text">
         <string>Удалить элемент с конца</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="list_dell">
        <property name="text">
         <string>Удалить элемент с начала</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>130</x>
      <y>30</y>
      <width>121</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Результаты запросов</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1268</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
